### ISSUE TITLE
Typechecker doesn’t play nice with eslint-plugin-markdown

### ISSUE THREAD

**Issue by remcohaszing** [typescript-eslint/typescript-eslint#2373]
When configuring `parserOptions.project` in combination with [eslint-plugin-markdown](https://github.com/eslint/eslint-plugin-markdown) v2, a parsing error is displayed and no linting rules are applied.

I tried disabling `parserOptions.project` by setting it to `null` or `[]` in overrides, but that didn’t work.

**Repro**

```JSON
{
  "overrides": [
    {
      "files": ["*.ts"],
      "parser": "@typescript-eslint/parser",
      "parserOptions": {
        "project": "tsconfig.json"
      }
    },
    {
      "files": ["**/*.md"],
      "plugins": ["markdown"],
      "processor": "markdown/markdown",
    },
  ],
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  },
  "parserOptions": {
    "...": "something"
  }
}
```

Markdown file content:

    ```ts
    function getRandomNumber(): Promise<number> {
      return 4;
    }

    console.log(getRandomNumber());
    ```

**Expected Result**

Something that is not a parser error.

Preferred specific solution: Make the rules work inside Markdown code blocks when using `eslint-plugin-markdown`
Fallback solution: Ignore files if they are processed using a custom processor.

**Actual Result**

```
README.md
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: examples/typescript-lib/README.md/2_2.ts.
The file must be included in at least one of the projects provided
```

**Additional Info**

A repository to reproduce the issue can be found [here](https://github.com/remcohaszing/eslint-config).

In `examples/typescript-lib/README.md`, change the `typescript` tagged code block to `ts` and run `yarn eslint .` to reproduce the issue.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `3.8.0` |
| `TypeScript`                | `3.9.7` |
| `ESLint`                    | `7.6.0` |
| `node`                      | `14.7.0` |
| `yarn`                       | `1.22.4` |
| `eslint-plugin-markdown`                       | `2.0.0-alpha.0` |

**Comment by bradzacher** [comment]
Nothing we can do about this.
TypeScript does the work in typechecked mode, and TS requires files on the disk.

the markdown plugin uses an eslint preprocessor to split the markdown file into a set of "pseudo-files" (hence the weird error with `README.md/2_2.ts`).
This file obviously will never exist, so TS cannot typecheck it.
